src/commands.ts

```diff
@@ -817,6 +837,9 @@ export function register(ctx: WorkspaceContext): vscode.Disposable[] {
         vscode.commands.registerCommand("swift.debugSnippet", () => debugSnippet(ctx)),
         vscode.commands.registerCommand("swift.runPluginTask", () => runPluginTask()),
         vscode.commands.registerCommand("swift.restartLSPServer", () => restartLSPServer(ctx)),
+        ...(ctx.swiftVersion.isGreaterThanOrEqual(new Version(6, 0, 0))
+            ? [vscode.commands.registerCommand("swift.reindexProject", () => reindexProject(ctx))]
```

Comment:
You can actually add a `"when"` clause to the package.json to hide the command under certain circumstances. Otherwise, the command will still show up in the command palette and give an ugly error message if the user tries to run it. Take a look at `contextKeys.ts` to see how to add a new context key. There are a bunch of examples for the `"when"` clause in the package.json already.

Reply:
Ideally we'd solve both these comments with a context key called `supportsReindexing`. It would be set by `languageClient.initializeResult.capatiblities.experimental['workspace/triggerReindex']` which would be returned when you initialize the language server. However this capability doesn't exist. 

@ahoppen Is this possible to add? If you agree I can work up a PR.